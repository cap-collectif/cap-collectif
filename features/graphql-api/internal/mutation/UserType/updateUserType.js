import '../../../_setup';

const UpdateUserTypeMutation = /* GraphQL */ `
  mutation UpdateUserTypesMutation($input: UpdateUserTypeInput!) {
    updateUserType(input: $input) {
      userType {
        id
        name
        translations {
          locale
          name
        }
        media {
          url
        }
      }
    }
  }
`;

describe('mutations.updateUserType', () => {
  const userTypeGlobalId = 'VXNlclR5cGU6NA=='; // UserType:4
  // the media:user-type-nonprofit and id:4 should be in the fixtures
  it('updates a user type', async () => {
    await expect(
      graphql(
        UpdateUserTypeMutation,
        {
          input: {
            id: userTypeGlobalId,
            translations: [
              {
                name: 'Institution avec media',
                locale: 'FR_FR',
              },
            ],
            media: 'user-type-nonprofit',
          },
        },
        'internal_admin',
      ),
    ).resolves.toMatchSnapshot();
  });

  // the id:4 should be in the fixtures
  it('tries to update a user type with a non-existing media', async () => {
    await expect(
      graphql(
        UpdateUserTypeMutation,
        {
          input: {
            id: userTypeGlobalId,
            translations: [
              {
                name: 'Institution avec media',
                locale: 'FR_FR',
              },
            ],
            media: 'this-media-does-not-exist',
          },
        },
        'internal_admin',
      ),
    ).rejects.toThrowError('This value is not valid.');
  });

  // the id:4 should be in the fixtures
  it('tries to update a user type with a basic user', async () => {
    await expect(
      graphql(
        UpdateUserTypeMutation,
        {
          input: {
            id: userTypeGlobalId,
            translations: [
              {
                name: 'Institution avec media',
                locale: 'FR_FR',
              },
            ],
          },
        },
        'internal_user',
      ),
    ).rejects.toThrowError('Access denied to this field.');
  });

  // the id:4 should be in the fixtures
  it('tries to update a user type without being authenticated', async () => {
    await expect(
      graphql(
        UpdateUserTypeMutation,
        {
          input: {
            id: userTypeGlobalId,
            translations: [
              {
                name: 'Institution avec media',
                locale: 'FR_FR',
              },
            ],
          },
        },
        'internal',
      ),
    ).rejects.toThrowError('Access denied to this field.');
  });
});
