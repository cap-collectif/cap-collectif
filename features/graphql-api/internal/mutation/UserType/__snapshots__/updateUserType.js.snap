// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations.updateUserType updates a user type 1`] = `
Object {
  "updateUserType": Object {
    "userType": Object {
      "id": "VXNlclR5cGU6NA==",
      "media": Object {
        "url": "https://capco.test/media/default/0001/01/nonprofit.jpg",
      },
      "name": "Institution avec media",
      "translations": Array [
        Object {
          "locale": "EN_GB",
          "name": "Institution",
        },
        Object {
          "locale": "FR_FR",
          "name": "Institution avec media",
        },
      ],
    },
  },
}
`;
